server/src/main/java/org/candlepin/dto/api/v1/GuestIdDTO.java

```diff
@@ -0,0 +1,219 @@
+import java.util.Map;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * A DTO representation of the GuestId entity
+ */
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.PROPERTY)
+@ApiModel(parent = TimestampedCandlepinDTO.class, description = "DTO representing a GuestIdDTO")
+@JsonFilter("GuestFilter")
+public class GuestIdDTO extends TimestampedCandlepinDTO<GuestIdDTO> {
+    public static final long serialVersionUID = 1L;
+
+    protected String id;
+    protected String guestId;
+    protected Map<String, String> attributes;
+
+    /**
+     * Initializes a new GuestIdDTO instance with null values.
+     */
+    public GuestIdDTO() {
+    }
+
+    /**
+     * Convenience constructor for tests with required parameter guestId
+     */
+    public GuestIdDTO(String guestId) {
+        this();
+        this.guestId = guestId;
+        attributes = new HashMap<String, String>();
+    }
+
+    /**
+     * Initializes a new GuestIdDTO instance which is a shallow copy of the provided
+     * source entity.
+     *
+     * @param source
+     *  The source entity to copy
+     */
+    public GuestIdDTO(GuestIdDTO source) {
+        super(source);
+    }
+
+    /**
+     * Retrieves the id field of this GuestIdDTO object.
+     *
+     * @return the id field of this GuestIdDTO object.
+     */
+    @HateoasInclude
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets the id to set on this GuestIdDTO object.
+     *
+     * @param id the id to set on this GuestIdDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public GuestIdDTO setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Retrieves the guestId field of this GuestIdDTO object.
+     *
+     * @return the guestId field of this GuestIdDTO object.
+     */
+    @HateoasInclude
+    public String getGuestId() {
+        return this.guestId;
+    }
+
+    /**
+     * Sets the guestId to set on this GuestIdDTO object.
+     *
+     * @param guestId the guestId to set on this GuestIdDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public GuestIdDTO setGuestId(String guestId) {
+        this.guestId = guestId;
+        return this;
+    }
+
+    /**
+     * Retrieves the attributes field of this GuestIdDTO object.
+     *
+     * @return the attributes field of this GuestIdDTO object.
+     */
+    @HateoasArrayExclude
+    public Map<String, String> getAttributes() {
+        return this.attributes != null ?
+            new MapView<String, String>(attributes) : null;
+    }
+
+    /**
+     * Sets the attributes on this GuestIdDTO object.
+     *
+     * @param attributes the attributes to set on this GuestIdDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public GuestIdDTO setAttributes(Map<String, String> attributes) {
```

Comment:
Add TODOs denoting the absence of the add/remove attribute methods and the intention to add them if we have a need in the future.

Reply:
done